homeassistant/components/logi_circle/camera.py

```diff
@@ -119,6 +123,28 @@ def device_state_attributes(self):
         """Return a still image from the camera."""
         return await self._camera.live_stream.download_jpeg()
 
+    async def handle_async_mjpeg_stream(self, request):
```

Comment:
Don't add mjpeg support, instead add our native stream support. You can literally just return the rtsp url and add `SUPPORT_STREAM` to the supported features and HA will do the rest.

Reply:
Gotcha. I need some guidance - I can only find examples of stream_source being a synchronous prop to the camera entity, but I need to call a coroutine in the API wrapper to get the URL (logi RTSP URLs are ephemeral).

Reply:
Ack, wrong again, won't work from the same loop.

Any suggestions @balloob? I'm not sure if it's kosher to spin up a new short-lived event loop just to call the `get_rtsp_url` coroutine from the API wrapper.

Reply:
So it was definitely an oversight on my part to make `stream_source` a property instead of an async method for exactly this reason.  It's on my list to change, but if you are feeling up to it, I'd welcome a separate PR doing just that across all integrations, which would make this PR easier on you.

Otherwise, you can request the stream source from the API in `async_setup_hass` and store it in a variable to return from the property, assuming the retrieved URL does not change or expire.

Reply:
The URL is short-lived unfortunately, if no connection is made the URL expires within about 10 seconds. Polling for URLs won't work in this instance, requesting the URL wakes battery-powered devices up from sleep so this approach will prevent any integrated cameras from sleeping.

I'm happy to take a crack at separate PR to add an async method for pulling the stream URL. Mind giving me some pointers on how this should be implemented? Specifically, should I replace all existing `stream_source` props with `async_stream_source` methods, or have it test for the existence of either the async method or sync prop?

Reply:
Yeah, start with making [this](https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/camera/__init__.py#L332) an async method versus a `@property`.  Then we will need to update everywhere that uses it by searching the code base for `stream_source` (some components are already being bad by doing I/O in a property).